--- contrib/restricted/googletest/googlemock/include/gmock/gmock-matchers.h	(b92e22f2baa7a4b746902dd4641a3025ca1f424f)
+++ contrib/restricted/googletest/googlemock/include/gmock/gmock-matchers.h	(working tree)
@@ -404,6 +404,7 @@ template <typename T, typename U>
 class MatcherCastImpl<T, Matcher<U>> {
  public:
   static Matcher<T> Cast(const Matcher<U>& source_matcher) {
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
     return Matcher<T>(new Impl(source_matcher));
   }
 
