--- a/googletest/include/gtest/internal/gtest-internal.h	(index)
+++ b/googletest/include/gtest/internal/gtest-internal.h	(working tree)
@@ -890,19 +890,16 @@ class GTEST_API_ Random {
 
 // HasDebugStringAndShortDebugString<T>::value is a compile-time bool constant
 // that's true if and only if T has methods DebugString() and ShortDebugString()
-// that return std::string.
 template <typename T>
 class HasDebugStringAndShortDebugString {
  private:
-  template <typename C>
-  static auto CheckDebugString(C*) -> typename std::is_same<
-      std::string, decltype(std::declval<const C>().DebugString())>::type;
-  template <typename>
+  template<typename C>
+  static std::true_type CheckDebugString(decltype(&C::DebugString));
+  template<typename>
   static std::false_type CheckDebugString(...);
 
-  template <typename C>
-  static auto CheckShortDebugString(C*) -> typename std::is_same<
-      std::string, decltype(std::declval<const C>().ShortDebugString())>::type;
+  template<typename C>
+  static std::true_type CheckShortDebugString(decltype(&C::ShortDebugString));
   template <typename>
   static std::false_type CheckShortDebugString(...);
 
