--- a/googletest/include/gtest/internal/gtest-port.h	(index)
+++ b/googletest/include/gtest/internal/gtest-port.h	(working tree)
@@ -393,8 +393,11 @@ typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
 #if GTEST_HAS_ABSL
 // When using Abseil, RE2 is required.
 #error #include "absl/strings/string_view.h"
-#include "re2/re2.h"
 #define GTEST_USES_RE2 1
+#endif
+
+#if GTEST_USES_RE2
+#include "re2/re2.h"
 #elif GTEST_HAS_POSIX_RE
 #error #include <regex.h>  // NOLINT
 #define GTEST_USES_POSIX_RE 1
@@ -866,17 +869,17 @@ GTEST_API_ bool IsTrue(bool condition);
 // char*` constructors.
 class GTEST_API_ RE {
  public:
-  RE(absl::string_view regex) : regex_(regex) {}                  // NOLINT
-  RE(const char* regex) : RE(absl::string_view(regex)) {}         // NOLINT
-  RE(const std::string& regex) : RE(absl::string_view(regex)) {}  // NOLINT
+  RE(std::string_view regex) : regex_(regex) {}                  // NOLINT
+  RE(const char* regex) : RE(std::string_view(regex)) {}         // NOLINT
+  RE(const std::string& regex) : RE(std::string_view(regex)) {}  // NOLINT
   RE(const RE& other) : RE(other.pattern()) {}
 
   const std::string& pattern() const { return regex_.pattern(); }
 
-  static bool FullMatch(absl::string_view str, const RE& re) {
+  static bool FullMatch(std::string_view str, const RE& re) {
     return RE2::FullMatch(str, re.regex_);
   }
-  static bool PartialMatch(absl::string_view str, const RE& re) {
+  static bool PartialMatch(std::string_view str, const RE& re) {
     return RE2::PartialMatch(str, re.regex_);
   }
 
