--- a/googletest/include/gtest/internal/gtest-death-test-internal.h    (index)
+++ b/googletest/include/gtest/internal/gtest-death-test-internal.h    (working tree)
@@ -226,6 +226,7 @@ inline Matcher<const ::std::string&> MakeDeathTestMatcher(
             #statement,                                                        \
             ::testing::internal::MakeDeathTestMatcher(regex_or_matcher),       \
             __FILE__, __LINE__, &gtest_dt)) {                                  \
+      /* NOLINTNEXTLINE(*-avoid-goto) */                                       \
       goto GTEST_CONCAT_TOKEN_(gtest_label_, __LINE__);                        \
     }                                                                          \
     if (gtest_dt != nullptr) {                                                 \
@@ -233,6 +234,7 @@ inline Matcher<const ::std::string&> MakeDeathTestMatcher(
       switch (gtest_dt->AssumeRole()) {                                        \
         case ::testing::internal::DeathTest::OVERSEE_TEST:                     \
           if (!gtest_dt->Passed(predicate(gtest_dt->Wait()))) {                \
+            /* NOLINTNEXTLINE(*-avoid-goto) */                                 \
             goto GTEST_CONCAT_TOKEN_(gtest_label_, __LINE__);                  \
           }                                                                    \
           break;                                                               \
--- a/googletest/include/gtest/internal/gtest-internal.h    (index)
+++ b/googletest/include/gtest/internal/gtest-internal.h    (working tree)
@@ -1403,6 +1403,7 @@ class NeverThrown {
     gtest_msg.value += " with description \"";                                 \
     gtest_msg.value += e.what();                                               \
     gtest_msg.value += "\".";                                                  \
+    /* NOLINTNEXTLINE(*-avoid-goto) */                                         \
     goto GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__);                \
   }
 
@@ -1426,12 +1427,14 @@ class NeverThrown {
       gtest_msg.value = "Expected: " #statement                             \
                         " throws an exception of type " #expected_exception \
                         ".\n  Actual: it throws a different type.";         \
+      /* NOLINTNEXTLINE(*-avoid-goto) */                                    \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__);           \
     }                                                                       \
     if (!gtest_caught_expected) {                                           \
       gtest_msg.value = "Expected: " #statement                             \
                         " throws an exception of type " #expected_exception \
                         ".\n  Actual: it throws nothing.";                  \
+      /* NOLINTNEXTLINE(*-avoid-goto) */                                    \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__);           \
     }                                                                       \
   } else //NOLINT                                                         \
@@ -1447,6 +1450,7 @@ class NeverThrown {
     gtest_msg.value += " with description \"";                    \
     gtest_msg.value += e.what();                                  \
     gtest_msg.value += "\".";                                     \
+    /* NOLINTNEXTLINE(*-avoid-goto) */                            \
     goto GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__); \
   }
 
@@ -1465,6 +1469,7 @@ class NeverThrown {
     GTEST_TEST_NO_THROW_CATCH_STD_EXCEPTION_()                           \
     catch (...) {                                                        \
       gtest_msg.value = "it throws.";                                    \
+      /* NOLINTNEXTLINE(*-avoid-goto) */                                 \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__);      \
     }                                                                    \
   } else                                                                 \
@@ -1484,6 +1489,7 @@ class NeverThrown {
       gtest_caught_any = true;                                       \
     }                                                                \
     if (!gtest_caught_any) {                                         \
+      /* NOLINTNEXTLINE(*-avoid-goto) */                             \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testanythrow_, __LINE__); \
     }                                                                \
   } else                                                             \
@@ -1507,6 +1507,7 @@ class NeverThrown {
         gtest_fatal_failure_checker;                                \
     GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement);      \
     if (gtest_fatal_failure_checker.has_new_fatal_failure()) {      \
+       /* NOLINTNEXTLINE(*-avoid-goto) */                           \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testnofatal_, __LINE__); \
     }                                                               \
   } else /* NOLINT */                                               \
