Workaround for 
https://github.com/llvm/llvm-project/issues/125353
--- contrib/restricted/googletest/googletest/include/gtest/gtest-printers.h	(b63e5757c322f6bff2f403b0fb65b7e0b7eaff47)
+++ contrib/restricted/googletest/googletest/include/gtest/gtest-printers.h	(7ee8ffa7c1c848f42486ddaccd23742638a35ac3)
@@ -829,6 +829,11 @@ void PrintTo(const ::std::pair<T1, T2>& value, ::std::ostream* os) {
   *os << ')';
 }
 
+template <typename C, typename D>
+void PrintTo(const ::std::chrono::time_point<C, D>& value, ::std::ostream* os) {
+	*os << "TimeStamp(" << static_cast<int64_t>(value.time_since_epoch().count()) << ")";
+}
+
 // Implements printing a non-reference type T by letting the compiler
 // pick the right overload of PrintTo() for T.
 template <typename T>
