--- contrib/libs/libpqxx/include/pqxx/internal/conversions.hxx	(index)
+++ contrib/libs/libpqxx/include/pqxx/internal/conversions.hxx	(working tree)
@@ -918,16 +918,6 @@ private:
 
 namespace pqxx
 {
-template<typename T> struct nullness<std::vector<T>> : no_null<std::vector<T>>
-{};
-
-
-template<typename T>
-struct string_traits<std::vector<T>>
-        : internal::array_string_traits<std::vector<T>>
-{};
-
-
 template<typename T> inline constexpr bool is_sql_array<std::vector<T>>{true};
 
 
